lib/src/pipeline/transform.dart

```diff
@@ -84,14 +84,22 @@ class TransformStep extends PipelineStep {
     return obj;
   }
 
+  static Float64List matrixDecoder(Iterable<num> nums) {
+    return Float64List.fromList(
+        nums.map((e) => e.toDouble()).toList(growable: false));
+  }
+
   static TransformStep deserialise(
     Map<String, dynamic> obj,
     PipelineStep? nextStep,
   ) {
-    Float64List initialMatrixStorage =
-        obj["initialMatrix"] ?? identityMatrixStorage;
-    Float64List finalMatrixStorage =
-        obj["finalMatrix"] ?? identityMatrixStorage;
+    Float64List? initialMatrixStorage = obj['initialMatrix'] != null
```

Comment:
create an extension serialization-deserialisation like how you did for textalignment